### PR TITLE
Remove workflow exit and update docs

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
The Workflow Exit was causing a build issue on Next repos. This was determined to be caused by the @kaizen/draft-modal components and its compilation in KAIO.

We have removed this component as it was an optional implementation for our consumers and have instead provided documentation on how to implement this within their solution.

There is also a minor change to the Wrapper to favor composability.

For more details on the issue, see the [KZ thread](https://cultureamp.slack.com/archives/C02NUQ27G56/p1685504632633019) where we discuss the problem

## What
- remove dep on @kaizen/draft-modal
- remove WorkflowExit component from Workflow pkg
  - Update reference in docs and provide implementation example
- Move Workflow Wrapper div into exportable component
  - Move styles into Wrapper folder

### CODE REVIEW

File: packages/components/src/Workflow/_docs/Workflow.mdx [hunk 1]
```diff
@@ -74,29 +72,44 @@ While the number of JSX element is not limited, is important to consider the rea
...
+  return (
+    <>
+      <Button
+        label={exitLabel}
+        icon={closeIcon}
+        iconPosition="end"
+        secondary
+        onClick={() => {
+          setShowModal(true)
+        }}
```
Comment (HeartSquared) [inline packages/components/src/Workflow/_docs/Workflow.mdx:92]:
```suggestion
        onClick={() => setShowModal(true)}
```

Just a suggestion, if you wanted to one-line it :)

File: packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  OverrideClassName<HTMLAttributes<HTMLDivElement>>
+>
+
+export const Wrapper = ({
+  children,
+  classNameOverride,
+  ...restProps
+}: WorkflowWrapperProps): JSX.Element => (
+  <div
+    className={classnames([styles.wrapper, classNameOverride])}
```
Comment (HeartSquared) [inline packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx:15]:
```suggestion
    className={classnames(styles.wrapper, classNameOverride)}
```
Array not necessary 😉

File: packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { PropsWithChildren, HTMLAttributes } from "react"
+import classnames from "classnames"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./Wrapper.module.scss"
+export type WorkflowWrapperProps = PropsWithChildren<
+  OverrideClassName<HTMLAttributes<HTMLDivElement>>
+>
+
```
Comment (mcwinter07) [inline packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx:8]: This update is less around the fix but a small DX improvement/continuity with the Main component we export that I thought we should provide

File: packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React, { PropsWithChildren, HTMLAttributes } from "react"
+import classnames from "classnames"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./Wrapper.module.scss"
+export type WorkflowWrapperProps = PropsWithChildren<
+  OverrideClassName<HTMLAttributes<HTMLDivElement>>
+>
```
Comment (HeartSquared) [inline packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx:7]:
```suggestion
export type WorkflowWrapperProps = OverrideClassName<HTMLAttributes<HTMLDivElement>> & {
  children: React.ReactNode
}
```

I'd probably why away from helper types as they can change between React versions. Better to be explicit (it's why we got rid of React.FC).

Comment (mcwinter07) [reply packages/components/src/Workflow/subcomponents/Wrapper/Wrapper.tsx:7]: Fair call, better to be explicit and safe

File: packages/components/src/Workflow/Workflow.tsx [hunk 5]
```diff
@@ -29,8 +28,8 @@ const WorkflowComponent = ({
   classNameOverride,
   ...restProps
 }: WorkflowProps): JSX.Element => (
-  <div
-    className={classnames([styles.workflowWrapper, classNameOverride])}
+  <Workflow.Wrapper
+    classNameOverride={classnames({ classNameOverride })}
```
[the remark this answers is not in the document]
Comment (mcwinter07) [reply packages/components/src/Workflow/Workflow.tsx:32]: Ah yes, completely unnecessary 😆

File: packages/components/package.json [hunk 6]
```diff
@@ -44,7 +44,6 @@
     "@kaizen/typography": "^2.3.10",
     "@kaizen/draft-tag": "^3.4.16",
     "@kaizen/brand": "^1.5.8",
-    "@kaizen/draft-modal": "^10.6.19",
```
Comment (HeartSquared) [inline packages/components/package.json:47]: Btw why aren't these alphabetical 😛

File: .changeset/lucky-zebras-serve.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+"@kaizen/components": minor
+---
+
+- remove dep on @kaizen/draft-modal
```
Comment (HeartSquared) [inline .changeset/lucky-zebras-serve.md:5]:
```suggestion
- remove dep on @kaizen/draft-modal as the styles causes compile to fail
```

Something like that so it doesn't sound like we're just removing it to go backwards :P

File: packages/components/package.json (hunk 6 above)
Comment (mcwinter07) [reply packages/components/package.json:47]: It's a secret alphabet... where "t" comes before "b"... Now you're in the inner circle 😈

File: packages/components/src/Workflow/Workflow.tsx [hunk 8]
```diff
@@ -29,8 +28,8 @@ const WorkflowComponent = ({
   classNameOverride,
   ...restProps
 }: WorkflowProps): JSX.Element => (
-  <div
-    className={classnames([styles.workflowWrapper, classNameOverride])}
+  <Workflow.Wrapper
+    classNameOverride={classnames(classNameOverride)}
```
Comment (HeartSquared) [inline packages/components/src/Workflow/Workflow.tsx:32]: Ah, I don't think I was clear before.

Either remove the whole line and have `classNameOverride` just be part of `...restProps`

```
  ...restProps
  }: WorkflowProps): JSX.Element => (
    <Workflow.Wrapper {...restProps}>  	
```

Or if you want to keep it destructured, then remove `classnames`

```
  classNameOverride,
  ...restProps
  }: WorkflowProps): JSX.Element => (
    <Workflow.Wrapper
      classNameOverride={classNameOverride}
      {...restProps}
    >  	
```

File: .changeset/lucky-zebras-serve.md (hunk 7 above)
Comment (mcwinter07) [reply .changeset/lucky-zebras-serve.md:5]: I hummed and had on that one, I'm never really sure how much context to put in a changeset vs what in the PR description. Happy to do so but feel like we need that doc on how best to write a changeset, least for our team

File: packages/components/src/Workflow/subcomponents/Wrapper/index.tsx [hunk 9]
```diff
@@ -0,0 +1 @@
+export * from "./Wrapper"
```
Comment (HeartSquared) [inline packages/components/src/Workflow/subcomponents/Wrapper/index.tsx:1]: Thoughts on renaming these `index.tsx` files to `index.ts` while you're here? 😄

File: .changeset/lucky-zebras-serve.md (hunk 7 above)
Comment (HeartSquared) [reply .changeset/lucky-zebras-serve.md:5]: Yep, that definitely came up in WoW as something we need to figure out. I guess I'd say it should list the change made and how it solves the problem? It's a fine line though, so we definitely need some sort of convention 😬

File: packages/components/src/Workflow/subcomponents/Wrapper/index.tsx (hunk 9 above)
Comment (mcwinter07) [reply packages/components/src/Workflow/subcomponents/Wrapper/index.tsx:1]: Could hurt to tackle that in one fell swoop

Comment (HeartSquared) [reply packages/components/src/Workflow/subcomponents/Wrapper/index.tsx:1]: 💔

File: packages/components/src/Workflow/Workflow.tsx (hunk 8 above)
Comment (mcwinter07) [reply packages/components/src/Workflow/Workflow.tsx:32]: Right, read you loud and clear now! I think I'll opt for the second one, to make it clear if anyone were to look at the src code

File: packages/components/src/Workflow/subcomponents/Wrapper/index.tsx (hunk 9 above)
Comment (mcwinter07) [reply packages/components/src/Workflow/subcomponents/Wrapper/index.tsx]: ** couldn't 😆 Should be good for another review now

File: packages/components/package.json (hunk 6 above)
Comment (mcwinter07) [reply packages/components/package.json:47]: Couldn't unsee it so I've fixed that now 😆

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**